fdbclient/BlobCipher.cpp

```diff
@@ -560,7 +603,10 @@ Reference<BlobCipherKey> BlobCipherKeyIdCache::insertBaseCipherKey(const Encrypt
 
 	Reference<BlobCipherKey> cipherKey =
 	    makeReference<BlobCipherKey>(domainId, baseCipherId, baseCipher, baseCipherLen, salt, refreshAt, expireAt);
-	keyIdCache.emplace(cacheKey, cipherKey);
+	auto result = keyIdCache.emplace(cacheKey, cipherKey);
+	ASSERT(result.second);
+	TraceEvent("InsertDone1").detail("R", result.second);
+
```

Comment:
Is it possible here: https://github.com/apple/foundationdb/pull/9845/files#diff-74240a17fd3787bf5b7b99522cd19290378f76c7f14efaf29be9ab8699dbe92cR588 that we return an expired key?

Reply:
Also do we need to make sure the cipher key returned there `isValid()`?

Reply:
> that we return an expired key?
No, as the call to `getLatestKey` would not return any expired key. 
Will add the check of cipher validity.